Entities/HeldDevice.cpp

```diff
@@ -566,11 +566,13 @@ void HeldDevice::DrawHUD(BITMAP *pTargetBitmap, const Vector &targetPos, int whi
 					const GameActivity *gameActivity = dynamic_cast<const GameActivity *>(activity);
 					if (gameActivity && gameActivity->GetViewState(viewingPlayer) == GameActivity::ViewState::ActorSelect) { unheldItemDisplayRange = -1.0F; }
 				}
-				// Note - to avoid item HUDs flickering in and out, we need to add a little leeway when hiding them if they're already displayed.
-				if (m_SeenByPlayer.at(viewingPlayer) && unheldItemDisplayRange > 0) { unheldItemDisplayRange += 3.0F; }
-				m_SeenByPlayer.at(viewingPlayer) = unheldItemDisplayRange < 0 || (unheldItemDisplayRange > 0 && g_SceneMan.ShortestDistance(m_Pos, g_CameraMan.GetScrollTarget(whichScreen), g_SceneMan.SceneWrapsX()).MagnitudeIsLessThan(unheldItemDisplayRange));
+				if (!m_SeenByPlayer.at(viewingPlayer)) {
```

Comment:
I tried for the longest time to figure out a solution that didn't require a new property flag lol 

If I had to, I would use a timer for the purpose

Reply:
> Why not just make it so if vel magnitude is > 2.0, you break out of this more cleanly? I know it's slightly different than your intent of making it so weapons don't show labels for a bit when they're dropped, but it's probably fine for weapons that are flying through the air to now show labels anyway.
I guess you're not a fan of this option?

And I get not wanting to make a new property but in general I don't think you need to be worried about that, it's not a big deal (especially since this one doesn't require ini or lua bindings) and it'd be a lot cleaner than it is now.

Reply:
I'm ultimately okay leaving this as-is though, I guess. I don't wanna deal with it haha